src/ipysim/widgets.py

```diff
@@ -26,54 +33,120 @@ def interactive_simulation(
+        T (float): Total simulation time in seconds.
+        dt (float): Time step for the simulation.
+        Kp_default (float): Default proportional gain for the PD controller.
+        Kd_default (float): Default derivative gain for the PD controller.
+        target_kp (float): Target proportional gain for evaluation.
+        target_kd (float): Target derivative gain for evaluation.
+
+    Returns:
+        None
     """
+    # # Suppress ODEintWarning
+    # warnings.filterwarnings("ignore", category=ODEintWarning)
+
+    global t, sol
     params = params or default_params
     state0 = state0 or default_state0
 
+    out = Output()
+    result_output = Output()
+
     def simulate_and_plot(Kp: float, Kd: float) -> None:
-        t, sol = simulate_maglev(Kp, Kd, T, dt, state0, params)
-
-        # This is not used?
-        # u_array = np.array([
-        #     -Kp * maglev_measurements(s, params["m"], params["mu0"])[0]
-        #     -Kd * maglev_measurements(s, params["m"], params["mu0"])[1]
-        #     for s in sol
-        # ])
-
-        plt.figure(figsize=(12, 5))
-
-        plt.subplot(1, 2, 1)
-        plt.plot(t, sol[:, 1], label='z (height)')
-        plt.plot(t, sol[:, 0], label='x (horizontal)')
-        plt.xlabel('Time [s]')
-        plt.ylabel('Position [m]')
-        plt.title('Position of levitating magnet')
-        plt.legend()
-        plt.grid(True)
-
-        plt.subplot(1, 2, 2)
-        plt.plot(sol[:, 0], sol[:, 2])
-        plt.xlabel('x')
-        plt.ylabel('theta')
-        plt.title('Phase plot: x vs theta')
-        plt.grid(True)
-
-        plt.tight_layout()
-        plt.show()
+        """
+        Simulate the maglev system and plot the results.
+
+        Args:
+            Kp (float): Proportional gain for the PD controller.
+            Kd (float): Derivative gain for the PD controller.
+
+        Returns:
+            None
+        """
+        global t, sol
+        try:
+            t, sol = simulate_maglev(Kp, Kd, T, dt, state0, params)
+
+            with out:
+                out.clear_output(wait=True)
+                plt.figure(figsize=(12, 5))
+                plt.subplot(1, 2, 1)
+                plt.plot(t, sol[:, 1], label='z (height)')
+                plt.plot(t, sol[:, 0], label='x (horizontal)')
+                plt.xlabel('Time [s]')
+                plt.ylabel('Position [m]')
+                plt.title('Position of levitating magnet')
+                plt.legend()
+                plt.grid(True)
+
+                plt.subplot(1, 2, 2)
+                plt.plot(sol[:, 0], sol[:, 2])
+                plt.xlabel('x')
+                plt.ylabel('theta')
+                plt.title('Phase plot: x vs theta')
+                plt.grid(True)
+
+                plt.tight_layout()
+                plt.show()
+
+        except Exception as e:
+            with out:
+                out.clear_output(wait=True)
+                print(f"Error: {e}")
```

Comment:
Bedre error handling her for brukere? Ikke bare printe feilen som ikke gir mening for brukere, kanskje?
```suggestion
            with out:
                out.clear_output(wait=True)
                print("An error occurred. Please try again.")
```